### PR TITLE
Array destructuring

### PR DESCRIPTION
**Pull request by kamil-tekiela**
Supercedes #145 and #451

I know that this duplicates the information from https://www.php.net/manual/en/function.list.php but I believe this information belongs on the array page. Unless we can somehow have a dedicated page for "array destructuring" that wouldn't be `list`-centric then I think having this information in both places is beneficial. Fewer people are using `list()` and nudging people to use the short syntax should be our goal. 

We could have a separate "book" for arrays, but it's also not really necessary. The page is long but not that long.

### CODE REVIEW

**Comment by ollieread** [comment]
Is there any news on this? I've just added issue #1753 to talk about adding an example of the short syntax to the `list()` entry page because it is just an alternate syntax.

**Comment by kamil-tekiela** [comment]
> The length of the page is not a problem, but duplicate information is. This is going out of sync almost certainly. If we need the same info on multiple pages, we should use one or more XML entities (I'm aware that the documentation guide mentions the rule of three here, but at least for long snippets, it makes sense to avoid doubles).
> 
> Do you think that is possible here?

I don't think this is possible. The pages have different structures, different examples and different text. There's nothing that can be extracted into a snippet. On a technical level, both syntaxes do the same, but they are still different syntaxes.

**Review by cmb69 — COMMENTED** [review]
Okay, I see. Then we should probably go ahead with this; after all, the array documentation page is no more messy than PHP's arrays are.

In the long run, I hope that the manual can be better structured. Still dreaming of integrating some stuff from the [php-langspec repo](https://github.com/php/php-langspec).

File: language/types/array.xml [hunk 1]
```diff
@@ -364,41 +364,6 @@ string(3) "foo"
...
-}
-
-$secondElement = getArray()[1];
-
-// or
-list(, $secondElement) = getArray();
-?>
-]]>
-    </programlisting>
-   </example>
```
[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply language/types/array.xml:384]: It was a year ago. I need to check it later and get back to you.

[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply language/types/array.xml:384]: It looks like I did it on purpose, but I have no idea why. I think I only wanted to remove `list(, $secondElement) = getArray();` but maybe I didn't read it properly and removed the whole example accidentally. I will bring it back, but without the last line.

File: language/types/array.xml [hunk 2]
```diff
@@ -364,41 +364,6 @@ string(3) "foo"
...
-   
-   <note>
-    <para>
-      Attempting to access an array key which has not been defined is
-      the same as accessing any other undefined variable:
-      an <constant>E_NOTICE</constant>-level error message will be
-      issued, and the result will be &null;.
-    </para>
-   </note>
```
[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply language/types/array.xml:393]: Yes, I moved it down, but I think it should be in both places. I will bring back the whole section. I should not have removed it.

**Review by cmb69 — APPROVED** [review]
Thank you!

**Pull request merged**